scripts/runHorusec.sh

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+# Check for install/updates at https://github.com/ZupIT/horusec
+
+source scripts/requireDocker.sh
+
+benchmark_version=$(scripts/getBenchmarkVersion.sh)
+horusec_version=$(docker run horuszup/horusec-cli horusec version 2>&1 | grep version | awk '{print $NF}')
+result_file="/src/results/Benchmark_$benchmark_version-horusec-$horusec_version.json"
+
+docker run \
+          -v /var/run/docker.sock:/var/run/docker.sock \
+          -v "$(pwd)":/src horuszup/horusec-cli \
+          horusec start -p /src -P "$(pwd)" -t 3600 \
+          -o="json" -O="$result_file"
```

Comment:
When I run this, this is what I get:
Dave.Wichers@USC02FJ2VXMD6T Benchmark % scripts/runHorusec.sh 
 
time="2021-09-16T18:57:03Z" level=info msg="Set log file to /tmp/horusec/horusec-log-2021-09-16 18:57:03.log"

time="2021-09-16T18:57:23Z" level=warning msg="{HORUSEC_CLI} When starting the analysis WE SKIP A TOTAL OF 280 FILES that are not considered to be analyzed. To see more details use flag --log-level=debug"

time="2021-09-16T19:00:20Z" level=warning msg="{HORUSEC_CLI} PLEASE DON'T REMOVE \".horusec\" FOLDER BEFORE THE ANALYSIS FINISH! Don’t worry, we’ll remove it after the analysis ends automatically! Project sent to folder in location: [/src/.horusec/ac920f08-6ad4-4096-9657-abaf04f9d702]"

time="2021-09-16T19:00:20Z" level=info msg="Hold on! Horusec is still analyzing your code. Timeout in: 3600s"
...
time="2021-09-16T19:05:50Z" level=info msg="Hold on! Horusec is still analyzing your code. Timeout in: 3270s"

And after it stopped, I looked for a results file in /results, and nothing new was there. Where does the result file go? Is it left on the Docker image and not exported to the local /results folder?

Reply:
What OS are you using? Just ran the script and the result file appeared in the results folder. (macOS 11.x)

Reply:
I'm on MacOS too. This time when I ran it, this happened: Benchmark % scripts/runHorusec.sh 
 
time="2021-09-18T15:31:08Z" level=info msg="Set log file to /tmp/horusec/horusec-log-2021-09-18 15:31:08.log"

time="2021-09-18T15:31:49Z" level=warning msg="{HORUSEC_CLI} When starting the analysis WE SKIP A TOTAL OF 32103 FILES that are not considered to be analyzed. To see more details use flag --log-level=debug"

time="2021-09-18T15:34:11Z" level=warning msg="{HORUSEC_CLI} PLEASE DON'T REMOVE \".horusec\" FOLDER BEFORE THE ANALYSIS FINISH! Don’t worry, we’ll remove it after the analysis ends automatically! Project sent to folder in location: [/src/.horusec/1a7af08d-2a75-4404-9fcb-550d633fdafc]"

time="2021-09-18T15:34:11Z" level=info msg="Hold on! Horusec is still analyzing your code. Timeout in: 3600s"
ERRO[0197] error waiting for container: invalid character 'u' looking for beginning of value 

I'm using Docker Desktop 3.6.0 w/Engine 20.10.0

Reply:
I upgraded to Docker Desktop 4.0.1 which has Engine 20.10.8 (I might have actually had Engine 20.10.8 before actually, not sure). Anyway, that didn't help.

Reply:
`error waiting for container: invalid character 'u' looking for beginning of value`

This sounds like a docker/mac issue. At least according to this:

https://github.com/docker/for-mac/issues/5139

Can you check this on another machine?

Reply:
I just pulled your changes and tried again and now get this??  Benchmark % scripts/runHorusec.sh
docker: Error response from daemon: Bad response from Docker engine.
See 'docker run --help'.

Reply:
So, I rebooted my Mac, and tried again. This time it ran but still nothing in /results. However, I did find this:  .horusec/a4fa4651-3281-43c4-87f2-47f45488ae75/result.json

And inside .horusec/a4fa4651-3281-43c4-87f2-47f45488ae75/. is a copy of ALL the files from the Benchmark project.  Is that the result file that is supposed to be moved to /results, and renamed of course?

Reply:
.horusec (and all within) should only exist temporary and be deleted after the run. Yes, this is a complete copy (comes from Horusec, not me!). There shouldn't be a need to manually move files since the flag `-O` (in combination with `-o`) should create the file in /results... 

Source: https://horusec.io/docs/cli/resources/#3-flags

I updated my docker, did `mvn clean`, deleted the image and pulled it again - but still it doen't fail and get the expected file in the expected directory. I'll try this on my work computer tomorrow (mac, too). Getting out of ideas though... 😅

Reply:
Added `--rm` to the script. Don't think this will make a difference but still it's cleaner (haha...)